Parsers/ASimWebSession/Parsers/ASimWebSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,197 @@
+Parser:
+  Title: Web Session ASIM parser for Cisco Firepower
+  Version: '0.1.0'
+  LastUpdated: Sep 04 2023
+Product:
+  Name: Cisco Firepower
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Cisco Firepower Documentation
+  Link: 
+    https://github.com/CiscoSecurity/fp-05-firepower-cli/tree/master/estreamer
+    https://www.cisco.com/c/en/us/td/docs/security/firepower/620/api/eStreamer/EventStreamerIntegrationGuide/IS-DCRecords.html#88027
+    https://www.cisco.com/c/en/us/td/docs/security/firepower/620/api/eStreamer/EventStreamerIntegrationGuide/IS-DCRecords.html#84248
+Description: |
+  This ASIM parser supports normalizing Cisco Firepower logs to the ASIM Web Session normalized schema.
+ParserName: ASimWebSessionCiscoFirepower
+EquivalentBuiltInParser: _ASim_WebSession_CiscoFirepower
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventFieldsLookup = datatable(
+      DeviceAction: string, 
+      DvcAction: string,
+      EventResult: string
+  )
+      [
+      "Detect", "Allow", "Partial",
+      "Block", "Deny", "Failure",
+      "Malware Cloud Lookup", "Deny", "Failure",
+      "Malware Block", "Deny", "Failure",
+      "Malware Allow List", "Allow", "Success",
+      "Cloud Lookup Timeout", "Deny", "Failure",
+      "Custom Detection", "Allow", "Partial",
+      "Custom Detection Block", "Deny", "Failure",
+      "Archive Block-Depth Exceeded", "Deny", "Failure",
+      "Archive Block-Encrypted", "Encrypt", "Failure",
+      "Archive Block-Failed to Inspect", "Deny", "Failure"
+  ];
+  let DirectionLookup = datatable (CommunicationDirection: string, NetworkDirection: string)[
+      "1", "Inbound",
+      "2", "Outbound"
+  ];
+  let parser=(disabled: bool=false) {
+      let alllogs=CommonSecurityLog
+          | where not(disabled) and DeviceVendor == "Cisco" and DeviceProduct == "Firepower"
+          | where DeviceEventClassID in("File:500:1", "FileMalware:502:1", "FireAMP:125:1")
+          | parse-kv AdditionalExtensions as (start: long) with (pair_delimiter=';', kv_delimiter='=');
+      let Fireamplogs = alllogs
+          | where DeviceEventClassID == "FireAMP:125:1"
+          | extend
+              EventMessage = DeviceCustomString5,
+              ThreatId = EventOutcome,
+              ThreatName = DeviceCustomString2
+          | extend AdditionalFields = bag_pack(
+                                  "policy",
+                                  DeviceCustomString1,
+                                  "process",
+                                  SourceProcessName,
+                                  "connectionInstance",
+                                  ProcessID,
+                                  "RequestClientApplication",
+                                  RequestClientApplication
```

Comment:
pls map this with HttpUserAgent
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/84cfa119-8aca-45dd-be92-ca7a2eea196b)

![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/c50709c9-5d19-457e-9b05-39f1e5ad65eb)

Reply:
Done.
As discussed, we are not getting proper values in the RequestClientApplication field in our environment but we have mapped it with HttpUserAgent as the field description suggests.